ingest/scripts/compare_hashes.py

```diff
@@ -22,13 +23,177 @@ class Config:
+
+
+@dataclass
+class SequenceUpdateManager:
+    submit: list[JointInsdcAccession]
+    revise: dict[JointInsdcAccession, Revision]
+    noop: dict[JointInsdcAccession, LoculusAccession]
+    blocked: dict[Status, dict[JointInsdcAccession, LoculusAccession]]
+    revoke: dict[
+        JointInsdcAccession, dict[LoculusAccession, InsdcAccession]
+    ]  # Map of new grouping joint insdc accessions to map of previous state
+    # i.e. loculus accessions (to be revoked) and their corresponding old joint insdc accessions
+    sampled_out: list[JointInsdcAccession]
+    hashes: list[float]
+
+    def __init__(self, config: Config):
+        self.config = config
+        self.submit = []
+        self.revise = {}
+        self.noop = {}
+        self.blocked = defaultdict(dict)
+        self.revoke = {}
+        self.sampled_out = []
+        self.hashes = []
+
+
+def notify(config: Config, text: str):
+    """Send slack notification with blocked revision details"""
+    if config.slack_hook:
+        requests.post(config.slack_hook, data=json.dumps({"text": text}), timeout=10)
+    logger.warn(text)
+
+
+def get_curated_fields(curated_data, pre_curated_data, current_state):
+    curated_fields = {}
+    added = {
+        k: curated_data[k]
+        for k in curated_data
+        if k not in pre_curated_data and curated_data[k] == current_state.get(k)
+    }
+    curated_fields.update(added)
+    removed = {
+        k: curated_data.get(k)
+        for k in pre_curated_data
+        if k not in curated_data and curated_data.get(k) == current_state.get(k)
+    }
+    curated_fields.update(removed)
+    modified = {
+        k: curated_data[k]
+        for k in pre_curated_data
+        if k in curated_data
+        and pre_curated_data[k] != curated_data[k]
+        and curated_data[k] == current_state.get(k)
+    }
+    curated_fields.update(modified)
+    return curated_fields
+
+
+def get_changed_fields(curated_data, pre_curated_data):
+    added = [k for k in curated_data if k not in pre_curated_data]
+    removed = [k for k in pre_curated_data if k not in curated_data]
+    modified = [
+        k for k in pre_curated_data if k in curated_data and pre_curated_data[k] != curated_data[k]
+    ]
+    return set(added + removed + modified)
+
+
+def handle_curation(
+    record: dict,
+    sorted_versions: list,
+    loculus_accession: str,
+    update_manager: SequenceUpdateManager,
+):
+    submitter_list = [sorted_version["submitter"] for sorted_version in sorted_versions]
+    curated_metadata_fields = {}
+    for index, submitter in enumerate(submitter_list):
+        if submitter != "insdc_ingest_user":
+            curated_data = sorted_versions[index]["original_metadata"]
+            pre_curated_data = sorted_versions[index - 1]["original_metadata"]
+            current_state = pre_curated_data = sorted_versions[-1]["original_metadata"]
+            curated_metadata_fields.update(
```

Comment:
reminder: also check if sequence has been curated - either add a sequence_hash for all entries (this will require db surgery to add to all existing entries) or need to get this information with get-original-metadata (lot of unnecessary sequence versions download from backend) or for every sequence ingest wants to revise call loculus (also a lot of requests)